core/domain/feedback_services.py

```diff
@@ -357,6 +358,31 @@ def pop_feedback_message_references(user_id, references_length):
         enqueue_feedback_message_email_task(user_id)
 
 
+def update_feedback_message_references(user_id, exploration_id):
+    """Removes feedback message references associated with exploration."""
+    if not user_id:
+        # If user is not logged in this will be True.
+        return
+
+    model = feedback_models.UnsentFeedbackEmailModel.get(user_id, strict=False)
+    if model is None:
+        # Model exists only if user has received feedback on exploration.
+        return
+
+    updated_references = []
+    for reference in model.feedback_message_references:
+        if exploration_id not in reference.values():
+            updated_references.append(reference)
```

Comment:
Ah, so you mean that we should only clear those thread messages which are opened by the user and keeping the rest. Yes, from user perspective this is more helpful  instead how it is implemented right now. As of now all messages get deleted without any regard that they have been seen or not. Deleting only opened threads makes more sense.